0.preprocess-sites/8.visualize-plate-layout.py

```diff
@@ -0,0 +1,392 @@
+    + gg.geom_point(gg.aes(fill="Threshold_FinalThreshold_Cells"), size=10)
+    + gg.geom_text(gg.aes(label="Site"), color="lightgrey")
+    + gg.facet_wrap("~Well")
+    + gg.coord_fixed()
+    + gg.theme_bw()
+    + gg.theme(
+        axis_text=gg.element_blank(),
+        axis_title=gg.element_blank(),
+        strip_background=gg.element_rect(colour="black", fill="#fdfff4"),
+    )
+    + gg.scale_fill_cmap(name="magma")
+)
+output_file = pathlib.Path(
+    output_figuresdir, "plate_layout_Cells_FinalThreshold_per_well.png"
+)
+cells_finalthresh_gg.save(output_file, dpi=300, verbose=False)
+
+# Plot final Nuclei thresholds per well
+nuclei_finalthresh_gg = (
+    gg.ggplot(image_df, gg.aes(x="x_loc", y="y_loc"))
+    + gg.geom_point(gg.aes(fill="Threshold_FinalThreshold_Nuclei"), size=10)
+    + gg.geom_text(gg.aes(label="Site"), color="lightgrey")
+    + gg.facet_wrap("~Well")
+    + gg.coord_fixed()
+    + gg.theme_bw()
+    + gg.theme(
+        axis_text=gg.element_blank(),
+        axis_title=gg.element_blank(),
+        strip_background=gg.element_rect(colour="black", fill="#fdfff4"),
+    )
+    + gg.scale_fill_cmap(name="magma")
+)
+output_file = pathlib.Path(
+    output_figuresdir, "plate_layout_Nuclei_FinalThreshold_per_well.png"
+)
+nuclei_finalthresh_gg.save(output_file, dpi=300, verbose=False)
+
+# Plot percent Confluent regions per well
+percent_confluent_gg = (
+    gg.ggplot(image_df, gg.aes(x="x_loc", y="y_loc"))
+    + gg.geom_point(gg.aes(fill="Math_PercentConfluent"), size=10)
+    + gg.geom_text(gg.aes(label="Site"), color="lightgrey")
+    + gg.facet_wrap("~Well")
+    + gg.coord_fixed()
+    + gg.theme_bw()
+    + gg.theme(
+        axis_text=gg.element_blank(),
+        axis_title=gg.element_blank(),
+        strip_background=gg.element_rect(colour="black", fill="#fdfff4"),
+    )
+    + gg.scale_fill_cmap(name="magma")
+)
+output_file = pathlib.Path(
+    output_figuresdir, "plate_layout_PercentConfluent_per_well.png"
+)
+percent_confluent_gg.save(output_file, dpi=300, verbose=False)
+
+# Create list of sites with confluent regions
+confluent_df = image_df.loc[image_df["Math_PercentConfluent"] > 0]
+confluent_df = (
+    confluent_df[["site", "Plate", "Well", "Site", "Math_PercentConfluent"]]
+    .sort_values(by=["site"])
+    .reset_index(drop=True)
+)
+if len(confluent_df.index) > 0:
+    confluent_output_file = pathlib.Path(
+        results_output, "sites_with_confluent_regions.csv"
+    )
+    confluent_df.to_csv(confluent_output_file)
+
+# Power Log Log Slope on Cell Painting images (proxy for focus)
+PLLS_df_cols = ["Plate", "Well", "Site"]
+PLLS_cols = []
+for name in painting_image_names:
+    PLLS_df_cols.append("ImageQuality_PowerLogLogSlope_" + name)
+    PLLS_cols.append("ImageQuality_PowerLogLogSlope_" + name)
+PLLS_df = image_df.loc[:, PLLS_df_cols]
+PLLS_df = PLLS_df.melt(id_vars=["Plate", "Well", "Site"], var_name="Channel").replace(
+    {"ImageQuality_PowerLogLogSlope_": ""}, regex=True
+)
```

Comment:
if we decide to go the route of a QC yaml file, I suggest making these a handful of short functions.

i.e. if and only if the QC metric column is chosen, than is the function executed as sort of a preprocessing step before visualization